prov/gni/src/gnix_nic.c

```diff
@@ -107,19 +119,43 @@ int _gnix_nic_free(struct gnix_nic *nic)
 	if ((nic->gni_cdm_hndl != NULL) && (v == 0))  {
 		if (nic->rx_cq_blk != NULL)
 			status = GNI_CqDestroy(nic->rx_cq_blk);
+			if (status != GNI_RC_SUCCESS)
+				GNIX_WARN(FI_LOG_EP_CTRL,
+					  "GNI_CqDestroy returned %s\n",
+					 gni_err_str[status]);
 		if (nic->rx_cq != NULL)
 			status = GNI_CqDestroy(nic->rx_cq);
+			if (status != GNI_RC_SUCCESS)
+				GNIX_WARN(FI_LOG_EP_CTRL,
+					  "GNI_CqDestroy returned %s\n",
+					 gni_err_str[status]);
 		if (nic->tx_cq_blk != NULL)
 			status = GNI_CqDestroy(nic->tx_cq_blk);
+			if (status != GNI_RC_SUCCESS)
+				GNIX_WARN(FI_LOG_EP_CTRL,
+					  "GNI_CqDestroy returned %s\n",
+					 gni_err_str[status]);
 		if (nic->tx_cq != NULL)
 			status = GNI_CqDestroy(nic->tx_cq);
+			if (status != GNI_RC_SUCCESS)
+				GNIX_WARN(FI_LOG_EP_CTRL,
+					  "GNI_CqDestroy returned %s\n",
+					 gni_err_str[status]);
 		status = GNI_CdmDestroy(nic->gni_cdm_hndl);
 		if (status != GNI_RC_SUCCESS) {
-			GNIX_ERR(FI_LOG_DOMAIN, "oops, cdm destroy failed\n");
+			GNIX_WARN(FI_LOG_EP_CTRL,
+				  "GNI_CdmDestroy returned %s\n",
+				  gni_err_str[status]);
 			ret = gnixu_to_fi_errno(status);
 			free(nic);
 		}
 
+		ret = gnix_mbox_allocator_destroy(nic->mbox_hndl);
```

Comment:
Isn't this going to lose a possible error status from above call to `GNI_CdmDestroy()`?  Or maybe we don't care about that assignment to `ret` above?